rain_api_core/urs_util.py

```diff
@@ -320,6 +321,7 @@ def do_login(args, context, cookie_domain='', aux_headers=None):
         else:
             redirect_to = get_base_url(context)
 
+        # TODO(reweeden): Why is the last check there if we're setting to the empty list anyways?
```

Comment:
we don't want to set it to empty list if it exists,

Reply:
Yea but it's doing this:
```python3
if not x:
    x = []
```

The only reason I can see for doing that is if you expect that the `user_groups` field could be equal to some other values that evaluate to false besides `[]`. Such as `False`, or `{}`. But that seems like it would be a bug or design flaw in the code that produces that dict (I think it comes from EDL), so I feel like this was just someone making a logic mistake and not realizing that the check is redundant.

Reply:
`get_profile()` might/should do a better job of validating the return from EDL instead of it happening here.

Reply:
Ok, that makes sense to me